plugins/woocommerce-blocks/assets/js/blocks/product-gallery/frontend.tsx

```diff
@@ -45,51 +45,41 @@ const getImageIndex = ( context: ProductGalleryContext, imageId: string ) => {
 	return imagesIds.indexOf( imageId );
 };
 
-const getImageId = ( context: ProductGalleryContext, imageIndex: number ) => {
-	const imagesIds = getCurrentImages( context );
-
-	if ( imageIndex < 0 ) {
-		return imagesIds.at( 0 ) || '';
-	}
-
-	if ( imageIndex > imagesIds.length - 1 ) {
-		return imagesIds.at( -1 ) || '';
-	}
-
-	return imagesIds[ imageIndex ];
-};
-
 const disableArrows = (
 	context: ProductGalleryContext,
 	nextImageIndex: number
 ) => {
 	const imagesIds = getCurrentImages( context );
-	context.disableLeft = nextImageIndex === 0;
-	context.disableRight = nextImageIndex === imagesIds.length - 1;
+	context.disableLeft = nextImageIndex === 1;
+	context.disableRight = nextImageIndex === imagesIds.length;
 };
 
 const selectImage = (
 	context: ProductGalleryContext,
-	type: 'prev' | 'next' | 'current'
+	type: 'prev' | 'next' | 'current' | 'first'
```

Comment:
Im curious why we have `first` and not `last`? Also if we're able to use `Math.min` to calculate `last` why can't we do similar to programmatically calculate `first`?

Reply:
> Im curious why we have first and not last?

There's no use case for `last`. We move to `first` when closing the dialog. This is because the image set is bigger in dialog than regular gallery which we'll change when revisiting thumbnails. But for now I suggest falling back to first image. Once we display all images in regular gallery we can easily stay on the same image without additional logic.

> Also if we're able to use Math.min to calculate last why can't we do similar to programmatically calculate first?

I'm using Math.min and Math.max for next/previous buttons so we don't move outside of image ranges. I'm not sure how that applies to `first` when we now explicitly what first is? Or you mean something else?

Reply:
I removed the `first` but introduced special `closeDialog` case. It's temporary until we have different number of images in "both" galleries. I wanted it to be more explicit. Also, renamed the variable to show better it's one-based.

https://github.com/woocommerce/woocommerce/pull/54570/commits/54a2ae6c20ddc32a687cf7a1124679ec02bfc019